tests/ttnn/unit_tests/gtests/accessor/kernels/reader_reshard.cpp

```diff
@@ -6,26 +6,31 @@
 #include "accessor/sharded_accessor.h"
 
 void kernel_main() {
-    const uint32_t bank_base_address = get_arg_val<uint32_t>(0);
+    const uint32_t bank_base_address = get_common_arg_val<uint32_t>(0);
 
     // The compile-time args are set up like this to highlight how you can use compile_time_args_skip
     // Recommended usage is to place the sequential compile-time args for distribution spec at the end
-    constexpr uint32_t rank = get_compile_time_arg_val(0);
-    constexpr uint32_t num_banks = get_compile_time_arg_val(1);
-    constexpr uint32_t base_idx = 2;
+    constexpr uint32_t base_idx_cta = 0;
+    constexpr uint32_t base_idx_crta = 1;
 
-    using input_dspec = distribution_spec_t<base_idx, rank, num_banks>;
-    constexpr uint32_t new_base_idx = base_idx + compile_time_args_skip<input_dspec>;
+    // using input_dspec = nd_sharding::distribution_spec_t<base_idx_cta, base_idx_crta>;
+    auto args = nd_sharding::make_args<base_idx_cta, base_idx_crta>();
+    constexpr uint32_t new_base_idx_cta = base_idx_cta + args.compile_time_args_skip();
+    uint32_t new_base_idx_crta = base_idx_crta + args.runtime_args_skip();
```

Comment:
```suggestion
    constexpr uint32_t new_base_idx_crta = base_idx_crta + args.runtime_args_skip();
```
For use second nd sharding tensor `new_base_idx_crta` should be `constexpr` for second args 
- `auto second_args = nd_sharding::make_args<new_base_idx_cta, new_base_idx_crta>();` 
Is it possible?

Reply:
Using only the `rank` as CTA is fine for now since it results in one binary per `rank`, but does the `bank` also need to be CTA?
If so, it seems like the number of binaries would increase significantly.

Reply:
it's number of banks, not banks, so this is one uint32_t. You just have to know how many banks you have to calculate offset in compile time

Reply:
> I’ll try tomorrow and see if it works.

Due to a sudden injury, I don’t think I’ll be able to check it today :(

Reply:
That's okay, I hope it's nothing serious

Reply:
Ohh sorry, number of builds yes, I thought about size of build.